controller/proxy-injector/server.go

```diff
@@ -76,24 +77,17 @@ func (w *WebhookServer) Shutdown() error {
 	return w.Server.Shutdown(context.Background())
 }
 
-func tlsConfig(certFile, keyFile string) (*tls.Config, error) {
-	certBytes, err := ioutil.ReadFile(certFile)
-	if err != nil {
-		return nil, err
-	}
-
-	keyBytes, err := ioutil.ReadFile(keyFile)
-	if err != nil {
-		return nil, err
-	}
+func tlsConfig(certFile, keyFile string, rootCA *ca.CA) (*tls.Config, error) {
+	dnsName := "linkerd-proxy-injector.linkerd.svc"
```

Comment:
This DNS name can change based on the namespace where Linkerd is installed. So it would be better to use the `controllerNamespace` variable that's passed into `NewWebhookServer` when constructing this string.

Reply:
Or better yet, you could use the `TLSIdentity` from `pkg/k8s. I haven't tried it, but I think something like this would work:

```diff
diff --git a/controller/proxy-injector/server.go b/controller/proxy-injector/server.go
index bdeb9c32..a67622a2 100644
--- a/controller/proxy-injector/server.go
+++ b/controller/proxy-injector/server.go
@@ -8,6 +8,7 @@ import (
 	"net/http"
 
 	"github.com/linkerd/linkerd2/controller/ca"
+	"github.com/linkerd/linkerd2/pkg/k8s"
 	pem "github.com/linkerd/linkerd2/pkg/tls"
 	log "github.com/sirupsen/logrus"
 	"k8s.io/client-go/kubernetes"
@@ -22,7 +23,7 @@ type WebhookServer struct {
 
 // NewWebhookServer returns a new instance of the WebhookServer.
 func NewWebhookServer(client kubernetes.Interface, resources *WebhookResources, addr, controllerNamespace, certFile, keyFile string, rootCA *ca.CA) (*WebhookServer, error) {
-	c, err := tlsConfig(certFile, keyFile, rootCA)
+	c, err := tlsConfig(certFile, keyFile, rootCA, controllerNamespace)
 	if err != nil {
 		return nil, err
 	}
@@ -77,8 +78,14 @@ func (w *WebhookServer) Shutdown() error {
 	return w.Server.Shutdown(context.Background())
 }
 
-func tlsConfig(certFile, keyFile string, rootCA *ca.CA) (*tls.Config, error) {
-	dnsName := "linkerd-proxy-injector.linkerd.svc"
+func tlsConfig(certFile, keyFile string, rootCA *ca.CA, controllerNamespace string) (*tls.Config, error) {
+	tlsIdentity := k8s.TLSIdentity{
+		Name:                "linkerd-proxy-injector",
+		Kind:                k8s.Service,
+		Namespace:           controllerNamespace,
+		ControllerNamespace: controllerNamespace,
+	}
+	dnsName := tlsIdentity.ToDNSName()
 	certAndPrivateKey, err := rootCA.IssueEndEntityCertificate(dnsName)
 
 	certPEM, err := pem.PEMEncodeCert(certAndPrivateKey.Certificate)
```

Reply:
yep, will certainly do :+1: